src/content/developers/docs/storage/index.md

```diff
@@ -6,7 +6,95 @@ sidebar: true
+
+This is the first type of persistence: **blockchain based** persistence.
+
+The issue with this, is that again, the chain could get far too big to feasibly upkeep and store all the data.
+
+The blockchain must also have some type of incentive structure. The following chains are generally paid at mining time - when the data is added to the chain, the nodes are paid to add the data on. 
+
+Platforms with blockchain based persistence:
+- Ethereum
+- [Arweave](https://www.arweave.org/)
+- [0Chain](https://0chain.net/)
+
+### Contract Based
+
+Contract based persistence has the intuition that data cannot be stored forever, and instead must be upkept with contract agreements. These are agreements made with multiple nodes that have promised to hold a piece of data for a period of time. They must be refunded or renewed whenever they run out to keep the data persisted. 
+
+Often, instead of storing all the data in a chain, they instead store the hash of where the data is located on a chain. This way, the entire chain doesn't need to store all the data, just a hash of where it's located. 
+
+Platforms with contract based persistence:
+- [Filecoin](https://docs.filecoin.io/about-filecoin/what-is-filecoin/)
+- [Skynet](https://siasky.net/)
+- [Stroj](https://storj.io/)
+
+### Additional considerations
+
+As a bonus, IPFS doesn't really have a incentive structure for holding data, but it is a great tool in the community. If you'd like data to persist on IPFS, you have to reach out to a pinning service, which will "pin" your data for you. 
+
+- [IPFS](https://ipfs.io/)
+
+## Data retention
+
+In order to retain data, systems must have some sort of mechanism to make sure data is retained.
+
+### Challenge mechanism
+
+One of the most popular ways to make sure data is retained, is to use some type of cryptographic challenge that is issued to the nodes to make sure they still have the data. A simple one is looking at Arweave's proof of access. They issue a challenge to the nodes to see if they have the data at both the most recent block and a random block in the past. If the node can't come up with the answer, they are penalized.
+
+Types of dStorage with a challenge mechanism:
+- 0Chain
+- Skynet
+- Arweave
+- Filecoin
+
+
+### Decentrality
+
+There isn't great tools to measure the level of decentralization of platforms but in general, you'll want to use tools that don't have some form of KYC to prove they are actually decentralized.
+
+Decentralized tools without KYC:
+- 0Chain (implementing a non-KYC edition)
+- Skynet
+- Arweave
+- Filecoin
+- IPFS
+- Ethereum
+
+### Consensus
+
+Most of these tools have their own version of a [consensus mechanism](/developers/docs/consensus-mechanisms/) but generally they are based on either [Proof of Work (PoW)](/developers/docs/consensus-mechanisms/pow/) or [Proof of Stake (PoS)](/developers/docs/consensus-mechanisms/pos/).
```

Comment:
Again, for consistency throughout the site, should we lowercase and hyphenate the proof-of-work/proof-of-stake links?